### PR TITLE
LF-4159 Create animal creation details views for animal batch

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This is the component to be imported alongside `<AnimalDetails />` as ExpandableContent for the `<AnimalsDetails />` step of the form. 
- [Storybook](http://localhost:6006/?path=/docs/components-addbatchdetails--docs)
- The same subcomponents are used by both Animal and Batch detail; Sayaka already has created the batch version of those stories -- they are nested under [Add Animal Details](http://localhost:6006/?path=/docs/components-addanimalsdetails--docs)
- Added in the disabling of the type and batch fields (and so removing the need to pass option arrays) as discussed this morning (Aug 12); updated Stories to provide a defaultValue for those two fields via form

@SayakaOno `<AnimalDetails />` was _so_ well organized! And all the subcomponents too for receiving the `BATCH` prop. We probably did not understand how well set up when we pointed this 😂 (or you had set the framework up after we had pointed)...

Jira link: https://lite-farm.atlassian.net/browse/LF-4159

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by kathyavini** [comment]
I removed the options from the type + breed ReactSelects in the `<General />` component as they weren't necessary in the context of the AddAnimalsDetails form... but realized belatedly that is it perhaps these exact 4 cards that we can re-use for the readonly + edit views. Is it so @Duncan-Brain? Then I will for sure restore. Maybe we can chat about the needs of readonly + edit in tech daily tomorrow?

**Comment by Duncan-Brain** [comment]
Yeah that was my understanding is that they will be re-used but yes, whoever gets that ticket may have to figure out some form structure stuff and fill us in!

**Comment by kathyavini** [comment]
@Duncan-Brain no I haven't even looked at LF-4168; I have my hands full with LF-4164. Let's discuss tomorrow because if neither LF-4168 nor this one have been merged yet and both are in review, maybe the extra fields need a separate ticket entirely since they are blocked by both.

File: packages/webapp/src/components/Animals/AddAnimalsDetails/type.ts [hunk 1]
```diff
@@ -55,9 +58,9 @@ export enum DetailsFields {
 }
 
 export type Option = {
-  [DetailsFields.TYPE]: ReactSelectOption<string>; // TODO: LF-4159
-  [DetailsFields.BREED]: ReactSelectOption<string>; // TODO: LF-4159
-  [DetailsFields.USE]: ReactSelectOption<number | string>; // TODO: LF-4159
+  [DetailsFields.TYPE]: AnimalSelectOption;
+  [DetailsFields.BREED]: AnimalSelectOption;
+  [DetailsFields.USE]: ReactSelectOption<number | string>;
```
Comment (SayakaOno) [inline packages/webapp/src/components/Animals/AddAnimalsDetails/type.ts:63]: I'm not sure why I didn't add this, but the value is a number, isn't it? 😓
Stories will need to be updated as well, sorry 🙏

**Review by SayakaOno — COMMENTED** [review]
Thank you so much for taking care of this!!! 🙏 

- Should we add "Batch" to the titles?
Screenshot 2024-08-15 at 4 23 00 PM

File: packages/webapp/src/components/Animals/AddAnimalsDetails/type.ts (hunk 1 above)
Comment (kathyavini) [reply packages/webapp/src/components/Animals/AddAnimalsDetails/type.ts]: Fixed, thank you!

**Comment by kathyavini** [comment]
> Should we add "Batch" to the titles?

Wow, I completely did not notice that extra word 👀 Thank you for your careful eyes again!

**Review by SayakaOno — APPROVED** [review]
Looks good, thank you!

**Review by Duncan-Brain — APPROVED** [review]
Looks good! 

(I personally liked the headings without "batch" -- but they are designed that way!) 

Only other thing I really see is you could consider fixing the Origin part of the batch story by passing `origin` to the Origin component similar to the Origin specific story.

**Comment by kathyavini** [comment]
> Only other thing I really see is you could consider fixing the Origin part of the batch story by passing origin to the Origin component similar to the Origin specific story.

@Duncan-Brain I like that idea! Because as we saw in tech daily, I was clueless about where the interactivity was set up and not 😄 Let's just put it on all the stories.

**Review by Duncan-Brain — APPROVED** [review]
Still works!

**Pull request merged**